.github/workflows/on-push-deploy.yml

```diff
@@ -0,0 +1,76 @@
+name: 'Run Deploy Actions on push to main trigger'
+
+on:
+  push:
+    branches:
+      - 'main'
+
+jobs:
+  run-tests-on-pull-request:
+    runs-on: [self-hosted, ubuntu, k8s]
```

Comment:
This needs to be ubuntu now as there are no self hosted runners anymore in this repo. (Also in the other workflow